Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+        {
+            var requestBuilder = new StringBuilder(EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Rates) + "prices?instruments=");
+
+            foreach (var instrument in instruments)
+            {
+                requestBuilder.Append(instrument.instrument + ",");
+            }
+            var requestString = requestBuilder.ToString().Trim(',');
+            requestString = requestString.Replace(",", "%2C");
+
+            var pricesResponse = MakeRequest<PricesResponse>(requestString);
+            var prices = new List<Price>();
+            prices.AddRange(pricesResponse.prices);
+
+            return prices;
+        }
+
+        /// <summary>
+        /// Posts an order on the given account with the given parameters
+        /// </summary>
+        /// <param name="requestParams">the parameters to use in the request</param>
+        /// <returns>PostOrderResponse with details of the results (throws if if fails)</returns>
+        public PostOrderResponse PostOrderAsync(Dictionary<string, string> requestParams)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders";
+            return MakeRequestWithBody<PostOrderResponse>(requestString, "POST", requestParams);
+        }
+
+        /// <summary>
+        /// Retrieves the list of open orders belonging to the account
+        /// </summary>
+        /// <param name="requestParams">optional additional parameters for the request (name, value pairs)</param>
+        /// <returns>List of Order objects (or empty list, if no orders)</returns>
+        public List<DataType.Order> GetOrderList(Dictionary<string, string> requestParams = null)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders";
+            var ordersResponse = MakeRequest<OrdersResponse>(requestString, "GET", requestParams);
+            var orders = new List<DataType.Order>();
+            orders.AddRange(ordersResponse.orders);
+            return orders;
+        }
+
+        /// <summary>
+        /// Cancels the order with the specified ID
+        /// </summary>
+        /// <param name="order">The order to cancel</param>
+        /// <returns>True if the request was made for the order to be canceled, false otherwise</returns>
+        public override bool CancelOrder(Order order)
+        {
+            Log.Trace("OandaBrokerage.CancelOrder(): " + order);
+            
+            if (!order.BrokerId.Any())
+            {
+                Log.Trace("OandaBrokerage.CancelOrder(): Unable to cancel order without BrokerId.");
+                return false;
+            }
+
+            foreach (var orderId in order.BrokerId)
+            {
+                CancelOrderAsync(orderId);
+                OnOrderEvent(new OrderEvent(order, "Oanda Cancel Order Event") { Status = OrderStatus.Canceled });
```

Comment:
Just reviewing this part of the code, I originally intended it to async (following Oanda's own implementation). Since then, I've started to move to code base for Oanda to not run Async, this is probably one part I missed.